### PR TITLE
Default instrumentation is added in absence of instrumentation config

### PR DESCRIPTION
**Pull request by dchappa**
*Issue #, if available:*
Pods would start without any instrumentation if there were no specification.

*Description of changes:*
Default instrumentation is applied to pods in absence of instrumentation config. Verified output on EKS cluster

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/instrumentation/podmutator.go [hunk 1]
```diff
@@ -21,7 +22,36 @@ import (
...
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "cloudwatch.aws.amazon.com/v1alpha1",
+			Kind:       "Instrumentation",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "java-instrumentation",
+			Namespace: "default",
+		},
+		Spec: v1alpha1.InstrumentationSpec{
+			Exporter: v1alpha1.Exporter{Endpoint: "http://amazon-cloudwatch-agent.amazon-cloudwatch:4317"},
```
Comment (lisguo) [inline pkg/instrumentation/podmutator.go:37]: nit: make `http://amazon-cloudwatch-agent.amazon-cloudwatch:4317` a const

File: pkg/instrumentation/podmutator.go [hunk 2]
```diff
@@ -21,7 +22,36 @@ import (
...
+			Propagators: []v1alpha1.Propagator{
+				v1alpha1.TraceContext,
+				v1alpha1.Baggage,
+				v1alpha1.B3,
+				v1alpha1.XRay,
+			},
+			Java: v1alpha1.Java{
+				Image: "160148376629.dkr.ecr.us-west-2.amazonaws.com/aws-apm-preview:latest",
```
Comment (lisguo) [inline pkg/instrumentation/podmutator.go:45]: nit: make `160148376629.dkr.ecr.us-west-2.amazonaws.com/aws-apm-preview:latest` a const

File: pkg/instrumentation/podmutator.go [hunk 3]
```diff
@@ -21,7 +22,36 @@ import (
...
+				Env: []corev1.EnvVar{
+					{Name: "OTEL_SMP_ENABLED", Value: "true"},
+					{Name: "OTEL_TRACES_SAMPLER_ARG", Value: "0.05"},
+					{Name: "OTEL_TRACES_SAMPLER", Value: "parentbased_traceidratio"},
+					{Name: "OTEL_EXPORTER_OTLP_TRACES_ENDPOINT", Value: "http://cloudwatch-agent.amazon-cloudwatch:4317"},
```
Comment (lisguo) [inline pkg/instrumentation/podmutator.go:50]: endpoint should be `http://amazon-cloudwatch-agent.amazon-cloudwatch:4317`

File: pkg/instrumentation/podmutator.go [hunk 4]
```diff
@@ -21,7 +22,36 @@ import (
...
+			Propagators: []v1alpha1.Propagator{
+				v1alpha1.TraceContext,
+				v1alpha1.Baggage,
+				v1alpha1.B3,
+				v1alpha1.XRay,
+			},
+			Java: v1alpha1.Java{
+				Image: "160148376629.dkr.ecr.us-west-2.amazonaws.com/aws-apm-preview:latest",
+				Env: []corev1.EnvVar{
+					{Name: "OTEL_SMP_ENABLED", Value: "true"},
```
Comment (lisguo) [inline pkg/instrumentation/podmutator.go:47]: all these should be consts too :) these might be changed in the future

**Review by lisguo — CHANGES_REQUESTED** [review]

File: pkg/instrumentation/podmutator.go [hunk 5]
```diff
@@ -19,9 +20,55 @@ import (
...
+	defaultNamespace                       = "default"
+	defaultKind                            = "Instrumentation"
+	otelSampleEnabledKey                   = "OTEL_SMP_ENABLED"
+	otelSampleEnabledDefaultValue          = "true"
+	otelTracesSamplerArgKey                = "OTEL_TRACES_SAMPLER_ARG"
+	otelTracesSamplerArgDefaultValue       = "0.05"
+	otelTracesSamplerKey                   = "OTEL_TRACES_SAMPLER"
+	otelTracesSamplerDefaultValue          = "parentbased_traceidratio"
+	otelExporterTracesEndpointKey          = "OTEL_EXPORTER_OTLP_TRACES_ENDPOINT"
+	otelExporterTracesEndpointDefaultValue = "http://amazon-cloudwatch-agent.amazon-cloudwatch:4317"
```
Comment (lisguo) [inline pkg/instrumentation/podmutator.go:37]: i guess we don't need this one since we will always have metrics/traces go through the same endpoint

**Review by lisguo — APPROVED** [review]

File: go.mod
Comment (mitali-salvi) [inline go.mod:1]: What are these go.mod changes for ?

Comment (dchappa) [reply go.mod:1]: These pretty much happened after syncing dependencies and running `go mod tidy`. I assume that they are not being used in this package anymore

Comment (mitali-salvi) [reply go.mod:1]: ack, thanks

**Review by mitali-salvi — APPROVED** [review]

**Pull request merged**